### PR TITLE
Use sysprop like with es.path.home to pass conf dir

### LINKED ISSUE #18689: Unxepected error when running command elasticsearch-plugin
**Elasticsearch version**:
5.0.0. alpha 3
**JVM version**:
1.8.0-92
**OS version**:
Ubuntu 3.16.0-71-lowlatency
**Description of the problem including expected versus actual behavior**:
Trying to install plugins with elasticsearch-plugin command, Checking new syntax with out any option or help gives error below:
**Steps to reproduce**:
 just run command

```
./elasticsearch-plugin
```

**Provide logs (if relevant)**:

```
/usr/share/elasticsearch/bin# ./elasticsearch-plugin
A tool for managing installed elasticsearch plugins

Commands
--------
list - Lists installed elasticsearch plugins
install - Install a plugin
remove - Removes a plugin from elasticsearch

Non-option arguments:
command

Option         Description
------         -----------
-h, --help     show help
-s, --silent   show minimal output
-v, --verbose  show verbose output
_**ERROR: E is not a recognized option**_

```

**Describe the feature**:

### PR DESCRIPTION
**Pull request by rjernst**
Currently we always pass -E to the the plugin cli with the conf dir, but
this causes a very confusing error message when not giving a specific
command to the plugin cli. This change makes path.conf pass just like
path.home. These are special settings, so passing via sysprops is the
right thing to do (it is all about how we pass between shell and java
cli).

closes #18689

### CODE REVIEW

File: distribution/src/main/resources/bin/elasticsearch-plugin [hunk 1]
```diff
@@ -82,9 +82,10 @@ HOSTNAME=`hostname | cut -d. -f1`
...
 declare -a args=("$@")
+PATH_PROPS="-Des.path.home=\"$ES_HOME\""
 
 if [ -e "$CONF_DIR" ]; then
-  args=("${args[@]}" -Edefault.path.conf="$CONF_DIR")
+  PATH_PROPS="$PATH_PROPS -Des.path.conf=\"$CONF_DIR\""
 fi
 
-exec "$JAVA" $ES_JAVA_OPTS -Delasticsearch -Des.path.home="$ES_HOME" -cp "$ES_HOME/lib/*" org.elasticsearch.plugins.PluginCli "${args[@]}"
+exec "$JAVA" $ES_JAVA_OPTS -Delasticsearch $PATH_PROPS -cp "$ES_HOME/lib/*" org.elasticsearch.plugins.PluginCli "${args[@]}"
```
Comment (jasontedor) [inline distribution/src/main/resources/bin/elasticsearch-plugin]: Unfortunately, this is not going to be quoting properly.

**Comment by rjernst** [comment]
@jasontedor I pushed a change to fix the quoting, and also made the equivalent change for windows bat script.

**Comment by jasontedor** [comment]
I think this looks right, and I tested it too. Can you clean up the merge conflicts and I'll give it a final review?

**Comment by rjernst** [comment]
@jasontedor I fixed the merge conflict.

**Comment by jasontedor** [comment]
LGTM

**Pull request merged**